website/data-storage/src/consumer/consumer.service.ts

```diff
@@ -88,9 +88,9 @@ export class ConsumerService {
 
   async programData(params: FindProgramParams): Result<Program> {
     try {
-      return (await this.programService.findProgram(params)) || { error: new ProgramNotFound().message };
-    } catch (error) {
-      return { error: error.message };
+      return (await this.programService.findProgram(params)) || { error: { message: new ProgramNotFound().message } };
```

Comment:
I think it's better to move throwing this error to `findProgram` method

Reply:
Sure; thanks.